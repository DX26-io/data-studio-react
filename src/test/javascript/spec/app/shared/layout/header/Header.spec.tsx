import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import Header from 'app/shared/layout/header/Header';

describe('Header tests', () => {
  const localeSpy = sinon.spy();
  const devProps = {
    isAuthenticated: true,
    isAdmin: true,
    currentLocale: 'en',
    onLocaleChange: localeSpy,
    ribbonEnv: 'dev',
    isInProduction: false,
    isSwaggerEnabled: true
  };
  const prodProps = {
    ...devProps,
    ribbonEnv: 'prod',
    isInProduction: true,
    isSwaggerEnabled: false
  };

  let mountedWrapper;

  const wrapper = (props = devProps) => {
    if (!mountedWrapper) {
      mountedWrapper = shallow(<Header {...props} />);
    }
    return mountedWrapper;
  };

  beforeEach(() => {
    mountedWrapper = undefined;
  });

  it('should render header component and have HTML header tag', () => {
    const component = wrapper();
    expect(component).toMatchSnapshot();
    expect(component.find('header').length).toEqual(1);
  });

  it('should hide the dev ribbon in prod', function() {
    const component = wrapper(prodProps);
    const ribbon = component.find('.ribbon.dev');
    expect(ribbon.length).toEqual(0);
  });

});
