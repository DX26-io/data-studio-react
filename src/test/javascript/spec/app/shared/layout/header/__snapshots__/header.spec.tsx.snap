// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests should render header component and have HTML header tag 1`] = `
<Fragment>
  <div
    className="ribbon dev"
  >
    <a
      href=""
    >
      <Translate
        component="span"
        contentKey="global.ribbon.dev"
      />
    </a>
  </div>
  <Connect(LoadingBar)
    className="loading-bar"
  />
  <ForwardRef($cb4e2cb36ae5e3ab01f62ab7d1672c67$var$View)
    backgroundColor="static-black"
    padding="size-150"
  >
    <header>
      <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
        justifyContent="space-between"
      >
        <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
          alignItems="center"
          justifyContent="center"
        >
          <Logo />
        </ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
        <ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)
          alignItems="end"
        >
          <Notifications />
          <DataStudioAvatar />
        </ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
      </ForwardRef($e9098d18abf2a118bc3bc95f0d402bb5$var$Flex)>
    </header>
  </ForwardRef($cb4e2cb36ae5e3ab01f62ab7d1672c67$var$View)>
</Fragment>
`;
